resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/VolcanoFeatureStep.scala

```diff
@@ -44,12 +44,8 @@ private[spark] class VolcanoFeatureStep extends KubernetesDriverCustomFeatureCon
 
   override def getAdditionalPreKubernetesResources(): Seq[HasMetadata] = {
     val client = new DefaultVolcanoClient
-
-    val template = if (kubernetesConf.isInstanceOf[KubernetesDriverConf]) {
-      kubernetesConf.get(KUBERNETES_DRIVER_PODGROUP_TEMPLATE_FILE)
-    } else {
-      kubernetesConf.get(KUBERNETES_EXECUTOR_PODGROUP_TEMPLATE_FILE)
-    }
+    // Currently only driver/job-level PodGroup is supported
+    val template = kubernetesConf.get(KUBERNETES_DRIVER_PODGROUP_TEMPLATE_FILE)
```

Comment:
This will work incorrectly because this class uses `with KubernetesExecutorCustomFeatureConfigStep`. I'd suggest to throw exception in the previous `if .. else ..` statement structure.

Reply:
This method not be called in executor as we know, but to protected something in future and more strictly, right?